packages/server/src/fhir/asyncjob.ts

```diff
@@ -0,0 +1,43 @@
+import { AsyncJob, Bundle, BundleEntry, BundleEntryResponse } from '@medplum/fhirtypes';
+import { Request, Response, Router } from 'express';
+import { asyncWrap } from '../async';
+import { Repository } from './repo';
+import { allOk } from '@medplum/core';
+import { sendResponse } from './routes';
+
+// Asychronous Job Status API
+// https://hl7.org/fhir/async-bundle.html
+
+export const asyncJobRouter = Router();
+
+asyncJobRouter.get(
+  '/:id/status',
+  asyncWrap(async (req: Request, res: Response) => {
+    const { id } = req.params;
+    const repo = res.locals.repo as Repository;
+    const asyncJob = await repo.readResource<AsyncJob>('AsyncJob', id);
+
+    if (asyncJob.status !== 'completed') {
```

Comment:
This is ok for now, but over time, I think we will want the ability to return progress notifications.  For example, in rebuilding or reindexing operations, give a percentage complete.

Reply:
Good call out! The plan would be to include `X-Progress` as part of the `202` response. https://build.fhir.org/async-bundle.html#3.2.6.2.4.0.1

Reply:
Oo, nice.